### PR TITLE
Add test for canonical paths

### PR DESCRIPTION
**Pull request by borrrden**
While at it, fixed a problem that it uncovered.  Verified these assertions on Windows, Linux, and Mac.

### CODE REVIEW

File: LiteCore/Support/FilePath.cc [hunk 1]
```diff
@@ -317,8 +317,8 @@ namespace litecore {
     string FilePath::canonicalPath() const {
 #ifdef _MSC_VER
         TempArray(wcanon, wchar_t, MAX_PATH);
-        const char* pathVal = path().c_str();
-	    const CA2WEX<256> wpath(pathVal, CP_UTF8);
+        string pathVal = path();
+	    const CA2WEX<256> wpath(pathVal.c_str(), CP_UTF8);
```
Comment (snej) [inline LiteCore/Support/FilePath.cc]: Still a glitch in the indentation of this line

File: LiteCore/tests/DataFileTest.cc [hunk 2]
```diff
@@ -416,6 +419,41 @@ N_WAY_TEST_CASE_METHOD (DataFileTestFixture, "DataFile Compact", "[DataFile]") {
...
+    string endPath = "C:\\subfolder\\";
+#else
+    auto tmpPath = FilePath::tempDirectory().path();
+    auto startPath = tmpPath + "folder/";
+    ::mkdir(startPath.c_str(), 777);
+    startPath += "../subfolder/";
+    auto endPath = tmpPath + "subfolder";
+    ::mkdir(endPath.c_str(), 777);
+#if __APPLE__
+    endPath = "/private" + endPath;
```
Comment (snej) [inline LiteCore/tests/DataFileTest.cc]: I don't believe this is true on iOS: tempDirectory() will be a path somewhere inside the app's data directory, not inside `/tmp`.

**Comment by snej** [comment]
Looks like you've modified `canonicalPath()` on master as well as in this PR?

**Comment by borrrden** [comment]
Doh yes. I meant to rebase this branch but o forgot. Will do that shortly.

**Pull request merged**